### PR TITLE
libisl: Make derivations generic, add 0.24

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Nothing in particular

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Could not build because my infra is not powerful enough to build this (my attempt tried to rebuild gcc10...)

### CODE REVIEW

File: pkgs/development/libraries/isl/0.17.1.nix [hunk 1]
```diff
@@ -1,24 +1,11 @@
 { lib, stdenv, fetchurl, gmp }:
```
Comment (r-burns) [inline pkgs/development/libraries/isl/0.17.1.nix:1]:
```suggestion
{ callPackage, lib, stdenv, fetchurl, gmp }:
```

Some of the older versions are missing this callPackage arg

Comment (matthiasbeyer) [reply pkgs/development/libraries/isl/0.17.1.nix]: whoops, that slipped. Thanks!

**Comment by matthiasbeyer** [comment]
I will rebase as soon as somebody approves this :smile:

File: pkgs/development/libraries/isl/0.24.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{ callPackage, lib, stdenv, fetchurl, gmp }:
```
Comment (r-burns) [inline pkgs/development/libraries/isl/0.24.nix:1]: I think this file needs to be named 0.24.0.nix

File: pkgs/development/libraries/isl/0.20.0.nix [hunk 3]
```diff
@@ -1,29 +1,16 @@
...
-  };
-
-  buildInputs = [ gmp ];
+{ callPackage, lib, stdenv, fetchurl, gmp }:
 
+callPackage ./generic.nix rec {
+  version = "0.20";
+  urls = [
+    "mirror://sourceforge/libisl/libisl-${version}.tar.xz"
+    "https://libisl.sourceforge.io/libisl-${version}.tar.xz"
```
Comment (r-burns) [inline pkgs/development/libraries/isl/0.20.0.nix:7]:
```suggestion
    "mirror://sourceforge/libisl/isl-${version}.tar.xz"
    "https://libisl.sourceforge.io/isl-${version}.tar.xz"
```

same goes for the others, I think

**Comment by r-burns** [comment]
The callPackage-inside-callPackage pattern here breaks overriding so stdenv won't build. Switching to `import ./generic.nix` fixes this. The changes are too big for an inline comment so I'll issue a PR against your fork to demonstrate.

**Comment by matthiasbeyer** [comment]
Merged your patches and rebase (+squashed my fixups) the branch on master. :tada:

**Comment by r-burns** [comment]
Awesome, exactly 1 rebuild so everything worked :)

Did you want keep `make derivation generic` and `add 0.24` as two separate commits? Otherwise I'll just squash-and-merge.

**Comment by matthiasbeyer** [comment]
I would merge it as-is, without any squashing because I think squashing just destroys history! :smiley:

**Comment by r-burns** [comment]
heh, well for a normal project I'd agree, but I think we're trying to squash when possible to keep the git history from growing out of control

**Comment by matthiasbeyer** [comment]
Well then, I'd say keep two separate commits for the generic making and one for the new version...

But you can squash your commits into mine if you want to.

Either way, I don't see any point in 


> keep the git history from growing out of control

But that should be discussed elsewhere 😂

**Comment by r-burns** [comment]
Fair enough, to be honest I didn't make that rule, I just don't want to get yelled at for bloating the history ;)

Anyway, this LGTM :)

**Comment by matthiasbeyer** [comment]
I think the "push to master" people bloat the history way more! ... I've been yelled at for that statement several times, yet I am not tired of making it again and again :laughing:

**Pull request merged**